# PIAIC_ISLAMABAD_BATCH_3/ARTIFICIAL INTELLIGENCE
###### Name: Tanzeel Ahmad
###### Roll No: PIAIC73909
###### Course: AIC
###### Class Timing: 06:45 PM to 09:45 Pm
###### Quarter: Q1
###### Batch 3
###### Date: 29-11-2019

In [1]:
import this

The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!


**When we assign values to some veriable after that we want to use that value and its address
then, Here we use deep value and shello value techniques!**

###### Deep Value and Shello value

In [2]:
a=2
b=2
print(a)
print(b)

print(id(a))
print(id(b))

8
8
140721171636848
140721171636848


We can access data directly by using its value and address with veriable name and id() built-in function respectively.Python provide the facility of Memory Management by placing same value that is assigned to different veriables at a single memory address whenever it is not changed.

Above example shows that veriables a and b have same vlaue that is "2" so this value is going to store at same memory address (140721171636848) in storage instead of different memory location that cause excess use of memory which is avoided in Python.

**Data Type**
To check data type of any veriale python has built-in function type(parameter)

In [3]:
a = 3
print(type(a))
a = '5'
print(type(a))
a = a + '5'    
                # Adding two values if both string then they concatinate
print(a)
                # If we want to convert one veriable data type to another called type casting
b = int(a) + 7
print(type(b))
print(b)

<class 'int'>
<class 'str'>
55
<class 'int'>
62


In [4]:
a = "Aalee Ahmad"
dir(a)
# Want to know which fuction/methods and attributes can be applied to string a

['__add__',
 '__class__',
 '__contains__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getitem__',
 '__getnewargs__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__len__',
 '__lt__',
 '__mod__',
 '__mul__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__rmod__',
 '__rmul__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'capitalize',
 'casefold',
 'center',
 'count',
 'encode',
 'endswith',
 'expandtabs',
 'find',
 'format',
 'format_map',
 'index',
 'isalnum',
 'isalpha',
 'isascii',
 'isdecimal',
 'isdigit',
 'isidentifier',
 'islower',
 'isnumeric',
 'isprintable',
 'isspace',
 'istitle',
 'isupper',
 'join',
 'ljust',
 'lower',
 'lstrip',
 'maketrans',
 'partition',
 'replace',
 'rfind',
 'rindex',
 'rjust',
 'rpartition',
 'rsplit',
 'rstrip',
 'split',
 'splitlines',
 'startswith',
 'strip',
 'swapcase',
 'title',
 'translate',
 'upper',


###### Operation can be applied by two types
1: In-Memory<br>
2: In-line

In [31]:
a = "aAlEe AhMaD"
print(a.upper()) # Inline Operation
print(a)

AALEE AHMAD
aAlEe AhMaD


In [32]:
a = "aAlEe AhMaD"
a = a.upper() # In Memory Operation
print(a)

AALEE AHMAD


In [33]:
a = "aAlEe AhMaD"
print(a.title())
print(a)

Aalee Ahmad
aAlEe AhMaD


In [34]:
a = "aAlEe AhMaD"
print(a.capitalize())
print(a)

Aalee ahmad
aAlEe AhMaD


In [10]:
a = "            aAlEe AhMaD           "
print(len(a))
print(a.strip())
b = len(a.strip())
print(b)
print('We saved ',len(a) - b,' Charecters Space/Memory ')

34
aAlEe AhMaD
11
We saved  23  Charecters Space/Memory 


In [11]:
c = a.split()
# Divide string into list where data is available in strig (Ignore spaces)
print(c)

['aAlEe', 'AhMaD']


In [12]:
d = " ".join(c)
# Join two or more list' elements
print(d)

aAlEe AhMaD


In [20]:
a = "You get what you work for, Not what you wish for!"
# Slicing : Same as indexing but access more than one vlaues
b = a[0:16:1]
# veriable[start:end:step] --> end is not included..
print(b)
print(a[::-1])
# Right to left slicing

You get what you
!rof hsiw uoy tahw toN ,rof krow uoy tahw teg uoY


In [30]:
# Search using index
print(a[a.index("work"):21])

work


In [35]:
a = "You get what you work for, Not what you wish for!"
a.find("work")

17

In [37]:
b = input("Search:")
print(a[a.index(b): a.index(b)+len(b)])
a[a.index(b): ]

Search:for
for


'for, Not what you wish for!'

In [50]:
# Multiline string
card = """
Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Course: AI
"""
print(card)


Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Course: AI



In [38]:
card = """
PIAIC ISLAMABAD
Bahria University 
Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
"""

print(card)


PIAIC ISLAMABAD
Bahria University 
Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz



In [40]:
student_name = "Tanzeel Ahmad"
f_name = "Mian Ahmad Nawaz"
fee = 3000
card = """
PIAIC ISLAMABAD
Bahria University 
Name: %s
Father's Name: %s
Fee : %d
"""
print(card%(student_name,f_name, fee))


PIAIC ISLAMABAD
Bahria University 
Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Fee : 3000



In [42]:
student_name = input("Name:")
f_name = input("Father's name: ")
fee = int(input("Fee"))
card = """
PIAIC ISLAMABAD
Bahria University 
Name: %s
Father's Name: %s
Fee : %d
"""
print(card%(student_name,f_name, fee))

Name:Tanzeel Ahmad
Father's name: Mian Ahmad Nawaz
Fee3000

PIAIC ISLAMABAD
Bahria University 
Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Fee : 3000



In [47]:
name = "Tanzeel Ahmad"
fname = "Mian Ahmad Nawaz"
fee = 3000

a=f"""
Bahria University Islamabad
Student Name: {name}
Father's Name: {fname}
Fee: {fee}
"""
print(a)


Bahria University Islamabad
Student Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Fee: 3000



In [45]:
name = "Tanzeel Ahmad"
fname = "Mian Ahmad Nawaz"
fee = 3000

a="""
Bahria University Islamabad
Student Name: {}
Father's Name: {}
Fee: {}
"""
print(a.format(name, fname,fee))


Bahria University Islamabad
Student Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Fee: 3000



In [48]:
name = "Tanzeel Ahmad"
fname = "Mian Ahmad Nawaz"
fee = 3000

a="""
Bahria University Islamabad
Student Name: {sn}
Father's Name: {fn}
Fee: {f}
"""
print(a.format(f=fee, fn=fname, sn=name))


Bahria University Islamabad
Student Name: Tanzeel Ahmad
Father's Name: Mian Ahmad Nawaz
Fee: 3000



In [54]:
name = "Tanzeel Ahmad"
fname = "Mian Ahmad Nawaz"
fee = 3000

print("Bahria University Islamabad \nStudent Name: " + name + "\nFather's Name:  \n"+fname+"\nFee: "+str(fee))

Bahria University Islamabad 
Student Name: Tanzeel Ahmad
Father's Name:  
Mian Ahmad Nawaz
Fee: 3000


# END!